app/src/main/java/unipd/se18/ocrcamera/PreProcessing.java

```diff
@@ -0,0 +1,369 @@
+     * @param imageMat the image we want to detect the brightness
+     * @return 0 if image is neither too bright nor too dark,
+     *         1 if image is too bright,
+     *         2 if image is too dark.
+     * @author Thomas Porro(g1), Giovanni Fasan(g1), Leonardo Pratesi(g1)
+     */
+    private int isBright(Mat imageMat){
+        //Converts the image into a matrix
+        Mat brightnessMat = new Mat();
+
+        //Changes the format of the matrix
+        Imgproc.cvtColor(imageMat, brightnessMat, Imgproc.COLOR_RGBA2RGB);
+
+        //Obtain 3 different matrix with the 3 elemental colors
+        List<Mat> color = new ArrayList<>();
+        Core.split(brightnessMat, color);
+
+        /*Each color is multiplied with his luminance.
+          For more informarions see https://en.wikipedia.org/wiki/Relative_luminance*/
+        Mat lumRed = new Mat();
+        Core.multiply(color.get(0), new Scalar(0.2126), lumRed);
+        Mat lumGreen = new Mat();
+        Core.multiply(color.get(1), new Scalar(0.7152), lumGreen);
+        Mat lumBlue = new Mat();
+        Core.multiply(color.get(2), new Scalar(0.0722), lumBlue);
+
+        //Sums the matrix of the colors into a single one
+        Mat lumTemp = new Mat();
+        Mat lum = new Mat();
+        Core.add(lumRed , lumGreen , lumTemp); //lumRed + lumGreen = lumTemp
+        Core.add(lumTemp , lumBlue , lum); //lumBlue + lumTemp = lum
+
+        //Calculate the sum of the values of all pixels
+        Scalar sum = Core.sumElems(lum);
+
+        //Image's bit
+        int bit;
+        switch ( brightnessMat.depth() ) {
+            case CV_8U:  bit = 8; break;
+            case CV_8S:  bit = 8; break;
+            case CV_16U: bit = 16; break;
+            case CV_16S: bit = 16; break;
+            case CV_32S: bit = 32; break;
+            case CV_32F: bit = 32; break;
+            case CV_64F: bit = 64; break;
+            default: return 0;
+        }
+        //Calculate the percentage of the brightness
+        double brightness = sum.val[0]/((Math.pow(2,bit)-1)*brightnessMat.rows()*brightnessMat.cols())*2;
+
+        Log.d(TAG, "Brightness:"+brightness);
+
+        /*Bounds to define if the image is dark or bright.
+          The values were decided on the basis of various tests*/
+        double upperBound = 0.9;
+        double lowerBound = 0.4;
+
+        if (brightness > upperBound){             //Image is too bright
+            return 1;
+        } else if (brightness < lowerBound){        //Image is too dark
+            return 2;
+        } else {      //Image is neither too bright nor too dark
+            return 0;
+        }
+    }
+
+
+    /**
+     * Change the brightness of the image into an optimal one
+     * @param image the image we want to modify the brightness
+     * @return the image with the modified brightness
+     * @author Thomas Porro(g1), Giovanni Fasan(g1), Oscar Garrido(g1)
+     */
+    private Bitmap editBright(Bitmap image){
```

Comment:
I suggest to call this method "adjustBrightness"

Reply:
Thank you for your suggestion. We will modify the name